### ISSUE TITLE
Modal Accessibility Issues

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2076]
The Modal Component has a few accessibility issues:

- A closed/unopened Modal should not have `role="dialog"`, that should be added when modal is opened and removed when modal is closed.
- Modal should have `aria-modal="true"` when opened (removed when closed)
- Modal should not have `aria-label` when closed.
- Modal should have `aria-hidden="true"` when closed
- Right now, the `aria-label` has to manually set and is listed as 'optional'. I think it'd be better if the value of the modal title was used automatically. The title would have an id (e.g. `id="contactFormModalTitle"`, and the modal would then have something like `aria-labelledby="contactFormModalTitle"` then no need for aria-label to be used at all (this is what Bootstrap does)
